lib/tilt/commonmarker.rb

```diff
@@ -2,53 +2,101 @@
 require_relative 'template'
 require 'commonmarker'
 
-aliases = {
-  :smartypants => :SMART
-}.freeze
-parse_opts = [
-  :FOOTNOTES,
-  :LIBERAL_HTML_TAG,
-  :SMART,
-  :smartypants,
-  :STRIKETHROUGH_DOUBLE_TILDE,
-  :UNSAFE,
-  :VALIDATE_UTF8,
-].freeze
-render_opts = [
-  :FOOTNOTES,
-  :FULL_INFO_STRING,
-  :GITHUB_PRE_LANG,
-  :HARDBREAKS,
-  :NOBREAKS,
-  :SAFE, # Removed in v0.18.0 (2018-10-17)
-  :SOURCEPOS,
-  :TABLE_PREFER_STYLE_ATTRIBUTES,
-  :UNSAFE,
-].freeze
-exts = [
-  :autolink,
-  :strikethrough,
-  :table,
-  :tagfilter,
-  :tasklist,
-].freeze
-
-
-Tilt::CommonMarkerTemplate = Tilt::StaticTemplate.subclass do
-  extensions = exts.select do |extension|
-    @options[extension]
-  end
+module Tilt
+  class CommonMarkerTemplate < StaticTemplate
+    ALIASES = {
+      :smartypants => :SMART
+    }.freeze
+    PARSE_OPTS = [
+      :FOOTNOTES,
+      :LIBERAL_HTML_TAG,
+      :SMART,
+      :smartypants,
+      :STRIKETHROUGH_DOUBLE_TILDE,
+      :UNSAFE,
+      :VALIDATE_UTF8,
+    ].freeze
+    RENDER_OPTS = [
+      :FOOTNOTES,
+      :FULL_INFO_STRING,
+      :GITHUB_PRE_LANG,
+      :HARDBREAKS,
+      :NOBREAKS,
+      :SAFE, # Removed in v0.18.0 (2018-10-17)
+      :SOURCEPOS,
+      :TABLE_PREFER_STYLE_ATTRIBUTES,
+      :UNSAFE,
+    ].freeze
+    EXTS = [
+      :autolink,
+      :strikethrough,
+      :table,
+      :tagfilter,
+      :tasklist,
+    ].freeze
+
+    V1_PARSE_OPTS = [
```

Comment:
We don't want to define different constants for commonmarker >= 1.0.  Ideally, we don't want to define constants at all.  Instead, the local variables previously defined should be changed if commonmarker > 1.0 is used.  You probably want an approach such as:

```ruby
if defined?(::Commonmarker)
  # new setting of parse_opts, render_opts, exts
  # new Tilt::CommonMarkerTemplate = Tilt::StaticTemplate.subclass do
else
  # previous setting of parse_opts, render_opts, exts
  # previous Tilt::CommonMarkerTemplate = Tilt::StaticTemplate.subclass do
end
```

Removing the private methods will also fix the failures in older Ruby versions (tilt supports back to Ruby 2.0).